kernel/src/simulationTools/Simulation.cpp

```diff
@@ -497,3 +497,19 @@ void Simulation::updateOutput(unsigned int)
   }
   DEBUG_END("Simulation::updateOutput()\n");
 }
+
+void Simulation::initializeOSIforDS(SP::OneStepIntegrator osi,
+                                    SP::DynamicalSystem ds,
+                                    SP::Model m, double time)
+{
+  // Keep OSI in the set, no effect if is already present.
+  insertIntegrator(osi);
+
+  // Associate the OSI to the DS in the topology.
+  _nsds->topology()->setOSI(ds, osi);
+
+  // If no Model, or OSI has no DSG yet, assume DS will be initialized
+  // later.  (Typically, during Simulation::initialize())
+  if (m && osi->dynamicalSystemsGraph())
+    osi->initializeDynamicalSystem(*m, time, ds);
+}
```

Comment:
I think it would make sense to have default argument for m and time, so that it is (imho) cleaner. The default time could come from the model. This brings also a question: when does time different from the currentTime in the Model?